TextToSpeech/TextToSpeechJsonRpc.cpp

```diff
@@ -166,17 +169,56 @@ namespace Plugin {
     {
         if(_tts) {
         Exchange::ITextToSpeech::Configuration config;
-        config.ttsEndPoint = GET_STR(parameters, "ttsendpoint", "");
-        config.ttsEndPointSecured = GET_STR(parameters, "ttsendpointsecured", "");
+        std::string proxyTtsEndPoint = GET_STR(parameters, "ttsendpoint", "");
+        std::regex urlPattern(R"((https?|ftp):\/\/\S+\/\S*)",std::regex::icase);
+        if(regex_match(proxyTtsEndPoint,urlPattern)) {
+            config.ttsEndPoint = proxyTtsEndPoint;
+        }
+        else {
+            config.ttsEndPoint = "";
+        }
+        std::string proxyTtsEndPointSecured = GET_STR(parameters, "ttsendpointsecured", "");
+        if(regex_match(proxyTtsEndPointSecured,urlPattern)) {
+            config.ttsEndPointSecured = proxyTtsEndPointSecured;
+        }
+        else {
+            config.ttsEndPointSecured = "";
+        }
         config.language = GET_STR(parameters, "language", "");
-        config.voice = GET_STR(parameters, "voice", "");
-        config.speechRate = GET_STR(parameters, "speechrate", "");
-        config.volume = (uint8_t) std::stod(GET_STR(parameters, "volume", "0.0"));
+        std::set<std::string> languageSet = {"en-US","es-MX","fr-CA","en-GB","de-DE","it-IT"};
```

Comment:
This reference list must either be populated from the configuration json file or should include all the standard languages